imap/mboxlist.h

```diff
@@ -196,6 +196,8 @@ int mboxlist_createmailboxcheck(const char *name, int mbtype,
 #define MBOXLIST_CREATE_SYNC                (1<<4)
 /* don't promote intermediary into a real mailbox */
 #define MBOXLIST_CREATE_KEEP_INTERMEDIARIES (1<<5)
+/* create mailbox as read only by everyone */
+#define MBOXLIST_CREATE_READ_ONLY           (1<<6)
```

Comment:
nit: the flag `MBOXLIST_CREATE_READ_ONLY`only is introduced here, but the previous two commits already use it. that would Cyrus leave in non-compilable state. Better to merge these three commits, as they really are one thing?